protocol/mqtt_paho/v2/message.go

```diff
@@ -88,14 +87,20 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
 			} else {
 				err = encoder.SetExtension(strings.TrimPrefix(userProperty.Key, prefix), userProperty.Value)
 			}
-		} else if userProperty.Key == contentType {
-			err = encoder.SetAttribute(m.version.AttributeFromKind(spec.DataContentType), string(userProperty.Value))
 		}
 		if err != nil {
 			return
 		}
 	}
 
+	contentType := m.internal.Properties.ContentType
```

Comment:
question: no changes needed for structured encoding?

Reply:
Yes! The binary message doesn't need the content type attribute.  Remove it now.

Reply:
We still need it here, cause the attribute `datacontenttype` of the event is decoded from this field with a binary message